# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

<div class="span5 alert alert-info">
### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value. Try using both the bootstrapping and the frequentist statistical approaches.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet
+ Formulas for the Bernoulli distribution: https://en.wikipedia.org/wiki/Bernoulli_distribution
</div>
****

In [2]:
import pandas as pd
import numpy as np
from scipy import stats

In [5]:
data = pd.io.stata.read_stata('C:/data/us_job_market_discrimination.dta')

In [7]:
#Exploring the data
data.head()

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
4  b  1          3       3        22       0          0         0         0   

   occupspecific    ...      compreq  orgreq  manuf  transcom  bankreal trade  \
0             17    ...          1.0     0.0    1.0       0.0       0.0   0.0   
1            316    ...          1.0     0.0    1.0       0.0       0.0   0.0   
2             19    ...          1.0     0.0    1.0       0.0       0.0   0.0   
3            313    ...          1.0     0.0    1.0       0.0       0.0   0.0   
4            313    ...          1.0     1.0    0.0       0.0       0.0   0.0   

  busservice othservice  missind  ownership  
0        0.0        0.0      0.0             
1        0.0        0.0      0.0             
2        0.0        0.0      0.0             
3        0.0        0.0      0.0             
4        0.0        1.0      0.0  Nonprofit  

[5 rows x 65 columns]

In [ ]:
#Seperate the data as needed for further analysis
w = data[data.race == 'w']
b = data[data.race == 'b']

<div class="span5 alert alert-success">
<p>Question 1:
    
We will use a hypothesis Z-Test to see if the proportions of callbacks differ by a significant amount. Since there are a large number of observations (n > 30) the CLT does apply.

Question 2: The Null Hypothesis: There is no significant difference between call backs for white and black sounding names.
w_prob = b_prob. 

Alternative Hypothesis: There is a significant difference between call backs for white and black sounding names. w_prob does not equal b_prob.

We will use a significance level of 0.05.
</p>
</div>

In [20]:
w = data[data.race=='w']
b = data[data.race=='b']


In [29]:
# Your solution to Q3 here
w_prob = (w.call.sum())/len(w)
b_prob = (b.call.sum())/len(b)
#Calculate relevant variables
mean = w_prob - b_prob
margin = np.sqrt((b_prob*(1 - b_prob)/(len(b))) + (w_prob*(1 - w_prob) /(len(w))))

(0.032032854209445585, 0.0077833705866767544)

In [40]:
#I've chosen to create a 99% confidence interval, for which z = 2.58
#Also, for our Ho, we are assuming the probabilities are equal, meaning the difference of means = 0
z = 2.58
confidence_interval = (mean - z*margin, mean + z*margin)
print("The 99% Confidence Interval is: ", confidence_interval)
new_z = (mean - 0)/margin
print("The p value in this test is: ", stats.norm.cdf(-new_z)*2)


The 99% Confidence Interval is:  (0.011951758095819557, 0.05211395032307162)
The p value in this test is:  3.862565207522622e-05


<div class="span5 alert alert-success">
<p> Question 4: In this experiment there is a 3.86*10-5 chance of there being no difference between the proportions of callbacks. With a 0.05 significance rate, and given that p < 0.05, we can reject the null hypothesis and state that there
is a significant difference between probability of being called back between white and black sounding names. 

Question 5: There is no way to conclusively say that race is the most important factor, especially when there are several other factors that could pertain to differences in callback rates. I think an analysis between race, and other factors such as honors, volunteer, years of experience would be neccesary to see if the impact of other relavant variables. Variables that are shown to be significant can then be tested in conjunction with others such as race, to further determine if race is truly the "most" important influential factor. For example, a test with 4 groups regarding all 4 distinct combinations of race and honors could be performed and analyzed to determine trends, and such an experiment could occur with many of the relevant variables in order to discover the biggest influential factor present in the data. 

</p>
</div>